### PR TITLE
fix: regex in setupDappMetadata function

### PR DESCRIPTION
**Pull request by omridan159**
fix the regex issue in the `setupDappMetadata` function in cases when the dapp url is set to     
`url: window.location.host`

### CODE REVIEW

File: packages/sdk/src/services/MetaMaskSDK/InitializerManager/setupDappMetadata.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ export function setupDappMetadata(instance: MetaMaskSDK) {
   const { options } = instance;
 
   // Check if iconUrl and url are valid
-  const urlPattern = /^https?:\/\//u; // Regular expression for URLs starting with http:// or https://
+  // eslint-disable-next-line require-unicode-regexp
+  const urlPattern = /^(http|https):\/\/[^\s]*|localhost(:\d+)?$/; // Regular expression for URLs starting with http:// or https://
```
Comment (abretonc7s) [inline packages/sdk/src/services/MetaMaskSDK/InitializerManager/setupDappMetadata.ts:19]: why do we need localhost?

Comment (abretonc7s) [reply packages/sdk/src/services/MetaMaskSDK/InitializerManager/setupDappMetadata.ts:19]: also can we fix it without the unicode regexp issue?

**Review by abretonc7s — CHANGES_REQUESTED** [review]
See comments + do we have tests to validate the function is behaving correctly?

File: packages/sdk/src/services/MetaMaskSDK/InitializerManager/setupDappMetadata.ts (hunk 1 above)
Comment (omridan159) [reply packages/sdk/src/services/MetaMaskSDK/InitializerManager/setupDappMetadata.ts:19]: On this case:
url: window.location.host

**Review by abretonc7s — APPROVED** [review]

**Pull request merged**